src/engine/controls/loopingcontrol.cpp

```diff
@@ -856,7 +856,16 @@ void LoopingControl::notifySeek(double dNewPlaypos) {
 void LoopingControl::setLoopingEnabled(bool enabled) {
     m_bLoopingEnabled = enabled;
     m_pCOLoopEnabled->set(enabled);
+    // TODO: QBasicAtomicInteger<T>::load() const is deprecated
+    // and should be replaced with QBasicAtomicInteger<T>::loadRelaxed()
+    // However, the proposed alternative has just been introduced in Qt
+    // 5.14. Until the minimum required Qt version of Mixx is increased,
+    // we need a version check here
+#if QT_VERSION >= QT_VERSION_CHECK(5, 14, 0)
```

Comment:
Unfortunately this is easily the most important fix of all, since the massive build log spam caused by this issue makes it impossible to spot warnings. I added a new commit https://github.com/mixxxdj/mixxx/pull/2423/commits/b591fc2f3b212ee2f866182d3afabb0c72c66f04 that adds a helper function. Another alternative could be to switch to `std::atomic` (which also has a `load()` method), but I don't know the implications. Grepping through the source shows that we already use it in some places:

```bash
$ grep -Hrn "std::atomic" .
./track/track.cpp:22:std::atomic<int> s_numberOfInstances;
./util/workerthread.h:126:    std::atomic<bool> m_suspend;
./util/workerthread.h:127:    std::atomic<bool> m_stop;
./util/workerthread.cpp:17:std::atomic<int> s_threadCounter(0);
./engine/engineworker.h:36:    std::atomic_flag m_notReady;
./test/globaltrackcache_test.cpp:66:    std::atomic<bool> m_stop;
./analyzer/analyzerthread.h:108:    // NOTE(uklotzde, 2018-01-04): Ideally we would use std::atomic<TrackPointer>,
```

Reply:
I guess there are only subtile differences:

QAtomicInt is guaranteed atomic while ```st::atomic<int>``` is not.
(It always is on our platforms)
Unfortunately there in c++ < 17 this can be only checked at runtime.

QAtomic does not support enums for some reasons while the standard version does.

QAtomic is copyable std:atomic not.

Note: Qt5 has removed the volatile nature of QAtomic. This means loadRelaxed is a normal integer read that can be moved around. 
We have to check if that is Ok in our code.

Reply:
The memory ordering doesn't matter for x86-64 where atomic loads always imply sequential memory ordering guarantees. But this assumption doesn't hold for other CPU architectures like ARM, so don't count on them.

Only rely on formal specifications! Not on guessing or on conclusions obtained by analyzing the actual implementation.

Reply:
This is not the case for QAtomicInt in Qt5 and loadRelaxed(). I have digged into the QT source and it turns out that it melds down to a simple inlined ```return value;``` statement. This change was done on purpose, to allow the compiler to join two consecutive loads.
It can be prevented by making the atomic volatile or using LoadAquire()
I guess this was the reason to deprecate the load() statement. It is a relaxed load in the QT case and a sequential load in the std case.